cli/types/cypress.d.ts

```diff
@@ -1421,6 +1417,29 @@ declare namespace Cypress {
      */
     off: Actions
 
+    /**
+     * Enables running Cypress commands in a secondary origin.
+     * @see https://on.cypress.io/origin
+     * @example
+     *    cy.origin('example.com', () => {
+     *      cy.get('h1').should('equal', 'Example Domain')
+     *    })
+     */
+    origin(urlOrDomain: string, fn: () => void): Chainable
```

Comment:
Since the return value of `fn` is [yielded](https://github.com/cypress-io/cypress/blob/890bde6a053525dfaaac9abcdf041c98753d50c6/packages/driver/cypress/integration/e2e/multi-domain/multi_domain_yield_spec.ts#L57-L61), shouldn't the signature be more like this?

```suggestion
    origin<T extends any>(urlOrDomain: string, fn: () => T): Chainable<T>
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/0c363dfeb095b20833a38d369e89924e229c4da1

Reply:
corresponding comment on docs PR https://github.com/cypress-io/cypress-documentation/pull/4439#discussion_r853435916